### PR TITLE
(sub)typesof

### PR DESCRIPTION
**Pull request by LetterN**
complain if i missed something

### CODE REVIEW

File: code/__DEFINES/__game.dm [hunk 1]
```diff
@@ -206,4 +206,8 @@ var/static/list/scarySounds = list('sound/weapons/thudswoosh.ogg','sound/weapons
 #define FONT_SIZE "5pt"
 #define FONT_COLOR "#09f"
 #define FONT_STYLE "Arial Black"
-#define SCROLL_SPEED 2
\ No newline at end of file
+#define SCROLL_SPEED 2
+
+//true and false
+#define TRUE 1
+#define FALSE 0
```
Comment (ninjanomnom) [inline code/__DEFINES/__game.dm]: ?

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -1638,3 +1638,7 @@ var/list/WALLITEMS = list(
 			error -= deltax
 	return line
 
+//hidden subtypes of
+/proc/subtypesof(typepath)
```
Comment (ninjanomnom) [inline code/__HELPERS/unsorted.dm]: `#define subtypesof(typepath) ( typesof(typepath) - typepath )`

File: code/__DEFINES/__game.dm (hunk 1 above)
Comment (LetterN) [reply code/__DEFINES/__game.dm]: is true and false already defined?

Comment (ninjanomnom) [reply code/__DEFINES/__game.dm]: It's defined by default in byond

Comment (LetterN) [reply code/__DEFINES/__game.dm]: @ninjanomnom removed it

File: code/__HELPERS/global_lists.dm [hunk 3]
```diff
@@ -93,79 +92,52 @@ var/global/list/backbaglist = list("Nothing", "Backpack", "Satchel")
 //////////////////////////
 
 /proc/makeDatumRefLists()
-	var/list/paths
+	// Hair - Initialise all /datum/sprite_accessory/hair into an list indexed by hair-style name
+	init_sprite_accessory_subtypes(/datum/sprite_accessory/hair, hair_styles_list, hair_styles_male_list, hair_styles_female_list)
+
+	// Facial Hair - Initialise all /datum/sprite_accessory/facial_hair into an list indexed by facialhair-style name
+	init_sprite_accessory_subtypes(/datum/sprite_accessory/facial_hair, facial_hair_styles_list, facial_hair_styles_male_list, facial_hair_styles_female_list)
```
Comment (ninjanomnom) [inline code/__HELPERS/global_lists.dm]: Could you take out code restructuring changes like this for now. They may be fine later but I'd prefer for large find and replace changes like this we keep the change as pure as possible

File: code/game/jobs/role_authority.dm [hunk 4]
```diff
@@ -20,74 +20,67 @@ var/global/datum/authority/branch/role/RoleAuthority
...
-			world << "<span class='debug'>Error setting up jobs, no job datums found.</span>"
-			log_debug("Error setting up jobs, no job datums found.")
-			return //No real reason this should be length zero, so we'll just return instead.
-
-		if(!squads_all.len)
-			world << "<span class='debug'>Error setting up squads, no squad datums found.</span>"
-			log_debug("Error setting up squads, no squad datums found.")
-			return
+/datum/authority/branch/role/New()
+	var/roles_all = subtypesof(/datum/job) - list(/datum/job/pmc, /datum/job/command, /datum/job/civilian, /datum/job/logistics, /datum/job/logistics/tech, /datum/job/marine, /datum/job/pmc/elite_responder)
```
Comment (ninjanomnom) [inline code/game/jobs/role_authority.dm]: Keep the list in multiline form

File: code/__HELPERS/global_lists.dm (hunk 3 above)
Comment (ninjanomnom) [reply code/__HELPERS/global_lists.dm]: Except fixing relative pathing, keep those kinds of changes

File: code/game/jobs/role_authority.dm [hunk 5]
```diff
@@ -20,74 +20,67 @@ var/global/datum/authority/branch/role/RoleAuthority
...
 
-		//TODO Come up with some dynamic method of doing this.
-		for(i in ROLES_REGULAR_ALL) //We're going to re-arrange the list for mode to look better, starting with the officers.
-			J = roles_for_mode[i]
+	//TODO Come up with some dynamic method of doing this.
+	for(i in ROLES_REGULAR_ALL) //We're going to re-arrange the list for mode to look better, starting with the officers.
+		J = roles_for_mode[i]
 			if(J) L[J.title] = J
-		roles_for_mode = L
+	roles_for_mode = L
```
Comment (ninjanomnom) [inline code/game/jobs/role_authority.dm]: This formatting here got real fucked up

**Comment by LetterN** [comment]
fuck, why did i use rebase, REE

**Comment by ninjanomnom** [comment]
Impressive how much you fucked it up

File: code/__HELPERS/global_lists.dm [hunk 6]
```diff
@@ -93,79 +92,69 @@ var/global/list/backbaglist = list("Nothing", "Backpack", "Satchel")
...
+			if(MALE)
 				hair_styles_male_list += H.name
+			if(FEMALE)
 				hair_styles_female_list += H.name
-
-	// Facial Hair - Initialise all /datum/sprite_accessory/facial_hair into an list indexed by facialhair-style name
-	paths = typesof(/datum/sprite_accessory/facial_hair) - /datum/sprite_accessory/facial_hair
-	for(var/path in paths)
+			else
+				hair_styles_list += H.name
```
Comment (ninjanomnom) [inline code/__HELPERS/global_lists.dm]: This isnt what the original loop did

File: code/__HELPERS/global_lists.dm [hunk 7]
```diff
@@ -93,79 +92,69 @@ var/global/list/backbaglist = list("Nothing", "Backpack", "Satchel")
...
 		switch(H.gender)
-			if(MALE)	facial_hair_styles_male_list += H.name
-			if(FEMALE)	facial_hair_styles_female_list += H.name
-			else
+			if(MALE)
 				facial_hair_styles_male_list += H.name
+			if(FEMALE)
 				facial_hair_styles_female_list += H.name
+			else
+				facial_hair_styles_list += H.name
```
Comment (ninjanomnom) [inline code/__HELPERS/global_lists.dm]: Same here

**Pull request merged**